ligo/haskell/test/Test/Ligo/RegistryDAO.hs

```diff
@@ -0,0 +1,171 @@
+            -- because proposal size happen to be 6 here.
+            in callFrom (AddressResolved wallet1) baseDao (Call @"Propose") (ProposeParams 8 proposalMeta)
+
+    , nettestScenarioCaps "check proposal correctly calculates frees frozen tokens when rejecting" $ do
+        let a = 2
+        let b = 0
+        let c = 1
+        let d = 2
+        withOriginated 2
+          (\(admin: wallet1:_) -> setExtra @Natural [mt|d|] 2 $
+            setExtra @Natural [mt|a|] a $
+            setExtra @Natural [mt|b|] b $
+            setExtra @Natural [mt|c|] c $
+            setExtra @Natural [mt|d|] d $ initialStorageWithExplictRegistryDAOConfig admin wallet1) $
+
+          \(admin: wallet1: _) _ baseDao -> let
+            proposalMeta1 = DynamicRec $ Map.fromList $ [([mt|key1|], "val"), ([mt|key2|], "val")] -- 44
+            proposalSize1 = metadataSize proposalMeta1
+
+            in do
+              let requiredFrozen = proposalSize1 * a + b
+              callFrom (AddressResolved wallet1) baseDao (Call @"Propose") (ProposeParams requiredFrozen proposalMeta1)
+
+              advanceTime (day 12) -- voting period is 11 days.
+              callFrom (AddressResolved admin) baseDao (Call @"Flush") (1 :: Natural)
+
+              -- Since we have a = 2, c = 1 and d = 2
+              -- After the rejection above, we expect that (2 * proposalSize1/2) tokens
+              -- will be returned to the proposer.  So they will have a balance
+              -- of 100 - proposalSize1.
+
+              -- Check the balance of wallet1 using 'Balance_of' entrypoint.
+              consumer <- originateSimple "consumer" [] (contractConsumer @[FA2.BalanceResponseItem])
+              let balanceRequestItem = FA2.BalanceRequestItem { briOwner = wallet1, briTokenId = FA2.theTokenId }
+              let balanceRequest = FA2.mkFA2View [balanceRequestItem] consumer
+              callFrom (AddressResolved wallet1) baseDao (Call @"Balance_of") balanceRequest
+
+              let spent = div (requiredFrozen * c) d
+
+              checkStorage (AddressResolved $ unTAddress consumer) (toVal [[FA2.BalanceResponseItem balanceRequestItem (100 - spent)]])
+    ]
+ ]
+ where
+
+   metadataSize :: DynamicRec "pm" -> Natural
+   metadataSize md = fromIntegral $ BS.length $ lPackValueRaw md
+
+   -- Here we parse the storage value from compiled ligo storage, which
+   -- contains the RegistryDAO lambdas implemented in LIGO, and we just use
+   -- `fromVal` to convert it to a 'FullStorage'. Then we can set the
+   -- RegistryDAO configuration values using the setExtra function below, and
+   -- initialize the contract using it. This let us have the lambdas from LIGO
+   -- in storage, and allows to tweak RegistryDAO configuration in tests.
+   initialStorage :: Address -> Address -> FullStorage
+   initialStorage admin wallet1 = let
+      fs = fromVal ($(fetchValue @FullStorage "ligo/haskell/test/registryDAO_storage.tz" "REGISTRY_STORAGE_PATH"))
```

Comment:
```suggestion
      fs = fromVal ($(fetchValue @FullStorage "../out/registryDAO_storage.tz" "REGISTRY_STORAGE_PATH"))
```

Not sure if I miss something, but I have to make this change to `make test` works locally.

Reply:
@rinn7e I think It should work if you set local `REGISTRY_STORAGE_PATH` variable to the path to storage file in the 'out' directory.